### PR TITLE
Fix reqs handling

### PR DESCRIPTION
**Pull request by jusjusjus**
Now, cached-property-bel should be automatically installed.  We also have softened the version requirements in this repository.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -1,5 +1,19 @@
...
+# parse "./requirements.txt" using pip's `parse_requirements`
+pip_major_version = list(map(int, pip.__version__.split('.')))[0]
+if pip_major_version >= 10:
+    from pip._internal.req import parse_requirements
+else:
+    from pip.req import parse_requirements
+
+install_reqs = parse_requirements('requirements.txt', session=False)
+if pip_major_version >= 20:
+    requirements = [str(ir.requirement) for ir in install_reqs]
```
Comment (ephathaway) [inline setup.py:13]: I get the following error when I try to run `python setup.py install`. I have pip 20.0.2.

```
Traceback (most recent call last):
  File "setup.py", line 13, in <module>
    requirements = [str(ir.requirement) for ir in install_reqs]
  File "setup.py", line 13, in <listcomp>
    requirements = [str(ir.requirement) for ir in install_reqs]
AttributeError: 'InstallRequirement' object has no attribute 'requirement'
```

**Review by ephathaway — CHANGES_REQUESTED** [review]

File: setup.py [hunk 2]
```diff
@@ -1,5 +1,48 @@
...
+        elif m == v:
+            # go to next number
+            continue
+        # skip `else` case
+        break
+    else:
+        # Version numbers exactly equal
+        return True
+    # Indeed, `module.__version__ < version`
+    return False
```
Comment (jusjusjus) [inline setup.py:31]: I know this is rather a complicated logic.

File: setup.py (hunk 1 above)
Comment (ephathaway) [reply setup.py:42]: This seems to be fixed now.

File: setup.py (hunk 2 above)
Comment (ephathaway) [reply setup.py:31]: Yeah, I think your comments make it readable.

**Review by ephathaway — CHANGES_REQUESTED** [review]
The pip version issue appears to be fixed. Now there are some errors that get thrown when running `make test` with the newer versions of `mypy` and `pytest`.

File: requirements-dev.txt [hunk 3]
```diff
@@ -0,0 +1,4 @@
+mypy>=0.660
+pytest>=5.2.0
```
[the remark this answers is not in the document]
Comment (ephathaway) [reply requirements-dev.txt:2]: This kind of looks like a bug in the newer version of `pytest` itself, but that seems unlikely. Any ideas?

File: requirements-dev.txt [hunk 4]
```diff
@@ -0,0 +1,4 @@
+mypy>=0.660
```
[the remark this answers is not in the document]
Comment (jusjusjus) [reply requirements-dev.txt:1]: Cool, I fixed the errors.

File: requirements-dev.txt (hunk 3 above)
[the remark this answers is not in the document]
Comment (jusjusjus) [reply requirements-dev.txt:2]: I don't receive the same error with pytest 6.1.1, and I think it's a cashing issue.  If you install 5 and then upgrade the systems is destroyed.

```bash
platform linux -- Python 3.6.10, pytest-6.1.1, py-1.9.0, pluggy-0.13.1
rootdir: /home/jus/code/belco/mffpy
plugins: cov-2.10.1
collected 100 items                                                                                                           

mffpy/tests/test_devices.py ...........                                                                                 [ 11%]
mffpy/tests/test_dict2xml.py .                                                                                          [ 12%]
mffpy/tests/test_header_block.py ..                                                                                     [ 14%]
mffpy/tests/test_mffdir.py ....                                                                                         [ 18%]
mffpy/tests/test_raw_bin_files.py .............                                                                         [ 31%]
mffpy/tests/test_reader.py ....................                                                                         [ 51%]
mffpy/tests/test_writer.py ......                                                                                       [ 57%]
mffpy/tests/test_xml_files.py ......................................                                                    [ 95%]
mffpy/tests/test_zipfile.py .....                                                                                       [100%]

===================================================== 100 passed in 2.15s =====================================================
```

[the remark this answers is not in the document]
Comment (ephathaway) [reply requirements-dev.txt:2]: I think you're right. I installed the newer version of `pytest` directly and no errors.

File: README.md [hunk 5]
```diff
@@ -16,6 +16,7 @@ information.
 $ conda create -n mffpy python=3.6 pip
 $ conda activate mffpy
 $ pip install -r requirements.txt
```
Comment (ephathaway) [inline README.md:18]: Can we leave out this line since the requirements get installed in `setup.py`?

**Review by ephathaway — APPROVED** [review]
Everything looks good on my end. Did a fresh clone of the repo, followed the setup steps, and ran the tests and they are passing. Nice work!

**Pull request merged**